### ISSUE TITLE
"uniq" is not exported by the List::Util module

### ISSUE THREAD

**Issue by stevendavis** [tseemann/prokka#414]
Seeing this error after a fresh install on Ubuntu.  What am I doing wrong?

$ bin/prokka
"uniq" is not exported by the List::Util module
Can't continue after import errors at bin/prokka line 31.
BEGIN failed--compilation aborted at bin/prokka line 31.

$ perl --version
This is perl 5, version 22, subversion 1 (v5.22.1) built for x86_64-linux-gnu-thread-multi

**Comment by tseemann** [comment]
List::Util is a core module but the uniq function was not provided until v1.45 which was included in an older version of Perl around 5.7.3 so I am unsure why this is happening if you have 5.22.

you can try `sudo cpan List::Util` to upgrade it. Or use `conda` or `brew` to manage your bioinformatics software.

**Comment by fluhus** [comment]
Hi,

I just ran into the same error. I installed Prokka using bioconda:

    conda install -c conda-forge -c bioconda -c defaults prokka

Running perl --version gives:

    This is perl 5, version 26, subversion 2 (v5.26.2) built for x86_64-linux-thread-multi

How did you end up solving this?

**Comment by stevendavis** [comment]
@fluhus I installed the following packages on Ubuntu:
`bioperl, cpanminus, libdatetime-perl,  libdigest-md5-perl, libxml-simple-perl`

**Comment by fluhus** [comment]
Thanks, @stevendavis!
I think I have all of those installed but I keep getting the error.

Any other ideas?

**Comment by fluhus** [comment]
I think I found the problem.

Googling the error message led me to [this thread](https://github.com/AOSC-Dev/aosc-os-abbs/issues/1679#issuecomment-469127848) which suggested that perl v5.28 is required. I checked and my installation (through conda) was v5.26.

I found [another perl](https://anaconda.org/biobuilds/perl) on conda which is v5.28. I did `conda remove perl` (which also removed prokka) and then installed that one. I reinstalled prokka and it worked.

So perhaps the instructions for conda should read something like:

    conda install -c biobuilds perl &&
    conda install -c bioconda prokka

And also point out that perl v5.28 or above is required.
